### PR TITLE
DF-2391 - Allow codebuild job to invoke lambda in some environments

### PR DESCRIPTION
**Pull request by bugs404**
- Add an optional variable that defaults to false and if true gives the codebuild job for the lambda function the ability to invoke the lambda function (for integration testing purposes)

### CODE REVIEW

File: lambda_function/ci.tf [hunk 1]
```diff
@@ -49,6 +49,7 @@ resource "aws_iam_role_policy" "codebuild" {
         "lambda:UpdateFunctionCode",
         "lambda:ListVersionsByFunction",
         "lambda:UpdateAlias"
+        "${var.codebuild_can_invoke_lambda ? ",lambda:InvokeFunction" : ""}"
```
Comment (bugs404) [inline lambda_function/ci.tf]: not sure if this is possible or I might have to break this out into separate statements and use the dynamic functionality.

Comment (bugs404) [reply lambda_function/ci.tf]: Testing in sandbox currently

**Review by eyablonowitz — CHANGES_REQUESTED** [review]
Please document this new option in the README and include an example of how it could be used.

File: README.md [hunk 2]
```diff
@@ -123,3 +124,11 @@ Ensure you add the following permissions to the lambda role
...
 
 []: https://www.terraform.io/docs/providers/aws/r/lambda_function.html
+
+## Codebuild and Integration Testing
+
+If invoking this module within an environment where Integration testing makes sense as part of CI, by setting the "codebuild_can_run_integration_test" argument to true
+ * The codebuild job that accompanies lambda ci is now able to invoke the lambda function
+ * The codebuild job will know if it's appropriate to perform integration testing in the environment it's running in according to env variable "run_integration_test"
+
+For an example implementation of a lambda-codebuild job setup to conditionally run integration tests see https://github.com/PatientPing/d2p_address_lookup_lambda
```
Comment (eyablonowitz) [inline README.md]: This is a public repo/module.  We don't want to put links to private repos in it.  But you could just copy https://github.com/PatientPing/d2p_address_lookup_lambda/blob/DF-2391/buildspec.yml#L16-L19 here to at least hint at how to wire this up.

**Review by eyablonowitz — CHANGES_REQUESTED** [review]

**Review by eyablonowitz — APPROVED** [review]
lgtm

**Pull request merged**